posts/2022-01-31-configuring-ocp-for-user-namespace.md

```diff
@@ -0,0 +1,198 @@
+  - [runc-1.0.3-992.rhaos4.10.el8.x86_64.rpm](https://ftweedal.fedorapeople.org/runc-1.0.3-992.rhaos4.10.el8.x86_64.rpm).
+  - [cri-o-1.23.0-990.rhaos4.10.git8c7713a.el8.x86_64.rpm](https://ftweedal.fedorapeople.org/cri-o-1.23.0-990.rhaos4.10.git8c7713a.el8.x86_64.rpm).
+
+> **Out of scope**: Build the runc and cri-o rpm packages.
+
+## Setting the configuration node
+
+- Clone freeipa-kustomize repository:
+
+  ```sh
+  git clone https://github.com/freeipa/freeipa-kustomize.git
+  ```
+
+- Retrieve the machine config pools by:
+
+  ```sh
+  oc get mcp
+  ```
+
+- Set the POOL environment variables with the names of the machine config pool
+  that are going to be configured:
+
+  ```sh
+  export POOL="worker"
+  ```
+
+  if you want to specify more than one:
+
+  ```sh
+  export POOL="worker master"
+  ```
+
+- Set the list of RPMs by:
+
+  ```sh
+  export RPM_PACKAGES="https://ftweedal.fedorapeople.org/runc-1.0.3-992.rhaos4.10.el8.x86_64.rpm https://ftweedal.fedorapeople.org/cri-o-1.23.0-990.rhaos4.10.git8c7713a.el8.x86_64.rpm"
+  ```
+
+- Now we just run:
+
+  ```sh
+  make -C config/static/nodes/userns configure
+  kustomize build config/static/nodes/userns | oc create -f -
+  ```
+
+- Monitor the node states by:
+
+  ```sh
+  watch oc get nodes
+  ```
+
+> It will take a few minutes (5-10minutes) as the configuration is applied node by node,
+> evict the node, restart the node, and make it avaiable. This
+> process is repeated for all impacted nodes. Eventually all the nodes will get a
+> Ready state and they could be used.
+
+
+## How is it structured?
+
+The main overlay at `config/static/nodes/usernd` is a copmosition of smaller
+ones, that are divided on:
+
+- `config/static/nodes/cgroup-v2`: Configure cgroup-v2 into the node, enabling
+  to mount cgroup v2 filessytem into the node.
+- `config/static/nodes/userns-subid`: Configure the necessary subid for the
+  user namespaces. Different files can be found at
+  `config/static/nodes/userns-subid/files` to spicify the subuid and subgid
+  information.
+  - `99-crio-userns.conf`: Enable the `io.kubernates.cri-o.userns-mode` annotation
+    into the PodSpec.
+  - `subuid` and `subgid`: Configure the subordinate ids to be used by the user namespace.
+- `config/static/nodes/rpm-overrides`: This configuration handle the RPM
+  package installation. This is made by creating a systemd unit, and executing
+  the command that install the RPM package. It is generated a resource for each
+  RPM and POOL. The package installation is checked before launch the RPM
+  command, so that future reboots does not try to install the RPM package
+  again. Here this is used for custom runc and cri-o rpm packages, but this
+  configuration could work for any RPM that we want to quickly test into our
+  OCP **development** cluster.
+
+## Checking that the configuration was applied
+
+- For the RPM packages:
+
+  ```sh
+  ssh -i ~/.crc/machines/crc/id_ecdsa core@192.168.130.11 runc --version
```

Comment:
Alternatively, use a node debug shell.  It is not to be assumed that SSH access will be possible.

Reply:
Thanks! I have added a note, and simplify the code blocks